libraries/AP_GyroFFT/AP_GyroFFT.cpp

```diff
@@ -829,28 +849,58 @@ uint8_t AP_GyroFFT::get_weighted_noise_center_frequencies_hz(uint8_t num_freqs,
 #endif
     }
 
-    const uint8_t tracked_peaks = MIN(_health, num_freqs);
     // pitch was good or required, roll was not, use pitch only
-    if (!_rpy_health.x || _harmonic_peak == FFT_HARMONIC_FIT_TRACK_PITCH) {
+    if (!_health.x || _harmonic_peak == FFT_HARMONIC_FIT_TRACK_PITCH) {
+        const uint8_t tracked_peaks = MIN(_health.y, num_freqs);
         for (uint8_t i = 0; i < tracked_peaks; i++) {
-            freqs[i] = get_slewed_noise_center_freq_hz(FrequencyPeak(i), 1);    // Y-axis
+            freqs[i] = get_noise_center_freq_hz(FrequencyPeak(i)).y;    // Y-axis
         }
         return tracked_peaks;
     }
     // roll was good or required, pitch was not, use roll only
-    if (!_rpy_health.y || _harmonic_peak == FFT_HARMONIC_FIT_TRACK_ROLL) {
+    if (!_health.y || _harmonic_peak == FFT_HARMONIC_FIT_TRACK_ROLL) {
+        const uint8_t tracked_peaks = MIN(_health.x, num_freqs);
         for (uint8_t i = 0; i < tracked_peaks; i++) {
-            freqs[i] = get_slewed_noise_center_freq_hz(FrequencyPeak(i), 0);    // X-axis
+            freqs[i] = get_noise_center_freq_hz(FrequencyPeak(i)).x;    // X-axis
         }
         return tracked_peaks;
     }
 
+    const uint8_t tracked_peaks = MIN(MAX(_health.x, _health.y), num_freqs);
     for (uint8_t i = 0; i < tracked_peaks; i++) {
-        freqs[i] = get_slewed_weighted_freq_hz(FrequencyPeak(i));
+        freqs[i] = get_weighted_freq_hz(FrequencyPeak(i));
     }
     return tracked_peaks;
 }
 
+// return whether there is a significant noise peak at the requested frequency
```

Comment:
comment a bit misleading, returning energy, not bool

Reply:
Fixed